### PR TITLE
chore: upgrade python to 3.11 :snake:  

### PR DESCRIPTION
**Pull request by aspicer**
> Python 3.11 is between 10-60% faster than Python 3.10. On average, we measured a 1.25x speedup on the standard benchmark suite.

'nuff said.

Steps to get this working are as such:
```
brew install python@3.11
rm -rf env
python3.11 -m venv env
. env/bin/activate
pip install uv
uv pip install -r requirements.txt -r requirements-dev.txt
```

**List of Changes**
- Code
    - Created a custom serializer to handle our `BooleanFields` with `null=True`. DRF changed the behavior for this, creating a null default if you don't pass it in from the frontend. We had been relying on the serializers to turn the nulls into Falses. This doesn't change behavior for existing models with Null (i think there is only one organization with `is_hipaa` set to null instead of False). https://github.com/encode/django-rest-framework/pull/8614 
    - DjangoRestFramework upgraded. This has a breaking router change because we register multiple viewsets with the same basename. Workaround implemented from here: https://github.com/nautobot/nautobot/pull/5546 
    - StrEnum is added (breaks `class Foo(str, Enum)`)
        - Change this in code and update the schema.py generator to return this 
        - https://github.com/python/cpython/issues/100458 
        - https://github.com/koxudaxi/datamodel-code-generator/issues/1313 
    - One small change to `ast.py` where `item_type: ConstantType = UnknownType()` became `item_type: ConstantType field(default_factory=UnknownType)`
- CI
    - `ci-backend.yml` checks out master and tries to migrate to it to test migrations, but this fails because of the version switch. Commented this out. **Comment this back in after merge**
    - Hadolint says `sudo` isn't okay in Dockerfile:35. Added ignore to the top of the file, as this docker is for dev and doesn't matter. https://github.com/PostHog/posthog/actions/runs/9517514426/job/26236279608?pr=22932

### CODE REVIEW

File: production.Dockerfile [hunk 1]
```diff
@@ -135,23 +135,20 @@ RUN apt-get update && \
...
     chmod -R 755 ./share/GeoLite2-City.mmdb
 
-
 #
 # ---------------------------------------------------------
 #
-# Build a version of the unit docker image for python3.10
-# We can remove this step once we are on python3.11
-FROM unit:python3.11 as unit
-FROM python:3.10-bullseye as unit-131-python-310
```
Comment (frankh) [inline production.Dockerfile:145]: Everything between here and line 236 can be removed now, and we can just use the unit:python3.11 image directly

File: production.Dockerfile [hunk 2]
```diff
@@ -236,7 +233,7 @@ CMD ["unitd", "--no-daemon", "--control", "unix:/var/run/control.unit.sock"]
 #
 # ---------------------------------------------------------
 #
-FROM unit-131-python-310
+FROM unit-132-python-311
 WORKDIR /code
```
Comment (frankh) [inline production.Dockerfile:237]: Just `FROM unit:python3.11` here

**Review by pauldambra — APPROVED** [review]
🚀 

looks like caching fun in the docker build

assuming the PR goes green we should use the new canary deploy and smash this in

**Comment by fuziontech** [comment]
looks like one other spot to bump the Python version
https://github.com/PostHog/posthog/blob/2f10422600704f6046b773c26cd876abf04032e4/Dockerfile#L86

**Comment by fuziontech** [comment]
wrt the hobby deploy - I'll fix that after this lands. Will probably require some attention.

**Pull request merged**